### PR TITLE
[Refactoring] rewrite InMemoryEventStore in Scala

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: event-sourcing/event-store-memory/src/main/scala/org/apache/james/eventsourcing/eventstore/memory/InMemoryEventStore.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import com.google.common.base.Preconditions
+import org.apache.james.eventsourcing.eventstore.{EventStore, History}
+import org.apache.james.eventsourcing.{AggregateId, Event}
+
+import scala.collection.JavaConverters._
+
+class InMemoryEventStore() extends EventStore {
+  private val storeRef: AtomicReference[Map[AggregateId, History]] = new AtomicReference(Map().withDefault(_ => History.empty()))
+
+  override def appendAll(events: util.List[Event]): Unit = if (!events.isEmpty) appendAll(events.asScala)
```
Comment (remk) [inline event-sourcing/event-store-memory/src/main/scala/org/apache/james/eventsourcing/eventstore/memory/InMemoryEventStore.scala]: events.nonEmpty

**Comment by Arsnael** [comment]
```
[ERROR] Errors: 
[63699c1ab4ee2433c29610b830015201f53b773e] [ERROR]   FileMailQueueTest.dequeueShouldBeConcurrent » ConditionTimeout Condition with ...
```
test this please

**Comment by chibenwa** [comment]
test this please

File: event-sourcing/event-store-memory/src/main/scala/org/apache/james/eventsourcing/eventstore/memory/InMemoryEventStore.scala (hunk 1 above)
Comment (mbaechler) [reply event-sourcing/event-store-memory/src/main/scala/org/apache/james/eventsourcing/eventstore/memory/InMemoryEventStore.scala]: it's a java List, there's no such method

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**